src/sentry/incidents/logic.py

```diff
@@ -105,6 +111,67 @@ def create_incident(
     return incident
 
 
+def calculate_incident_start(query, projects, groups):
+    params = {}
+    if groups:
+        params['issue.id'] = [g.id for g in groups]
+        end = max(g.last_seen for g in groups)
+    else:
+        end = timezone.now()
+
+    params['start'] = end - timedelta(days=7)
+    params['end'] = end
+
+    if projects:
+        params['project_id'] = [p.id for p in projects]
+
+    query_args = get_snuba_query_args(query, params)
+
+    result = raw_query(
+        aggregations=[
+            ('count()', '', 'count'),
+            ('min', 'timestamp', 'first_seen'),
+        ],
+        orderby='time',
+        groupby=['time'],
+        rollup=int(timedelta(minutes=30).total_seconds()),
```

Comment:
Should this bucket be smaller? Perhaps 5 or 10 minutes?

Reply:
Dropped it down to 15. I think that one improvement I could make is to make the rollup smaller, and use a sliding window across the counts, so that we kind of get a rolling average over a period of time. Might be a little more accurate, but we can see what product thinks